content/building/build-versioning.md

```diff
@@ -46,3 +46,112 @@ set -x
+3. Enter the name for the key and select an access level. We recommend choosing either `Developer` or `App Manager`, read more about Apple Developer Program role permissions [here](https://help.apple.com/app-store-connect/#/deve5f9a89d7).
+4. Click **Generate**.
+5. As soon as the key is generated, you can see it added in the list of active keys. Click **Download API Key** to save the private key for later. Note that the key can only be downloaded once.
+
+### Saving the API key to environment variables
+
+Save the API key and the related information as [environment](../getting-started/yaml#environment) variables. Make sure to [encrypt](./encrypting/#encrypting-sensitive-data) the values of the variables before adding them to the configuration file.
+
+```yaml
+environment:
+  vars:
+    APP_STORE_CONNECT_ISSUER_ID: Encrypted(...)
+    APP_STORE_CONNECT_KEY_IDENTIFIER: Encrypted(...)
+    APP_STORE_CONNECT_PRIVATE_KEY: Encrypted(...)
+```
+
+- `APP_STORE_CONNECT_KEY_IDENTIFIER`
+
+  In **App Store Connect > Users and Access > Keys**, this is the **Key ID** of the key.
+
+- `APP_STORE_CONNECT_ISSUER_ID`
+
+  In **App Store Connect > Users and Access > Keys**, this is the **Issuer ID** displayed above the table of active keys.
+
+- `APP_STORE_CONNECT_PRIVATE_KEY`
+
+  This is the private API key downloaded from App Store Connect. Note that when encrypting files via UI, they will be base64 encoded and would have to be decoded during the build. Alternativey, you can encrypt the **contents** of the file and save the encrypted value to the environment variable.
+
+{{<notebox>}}
+Alternatively, each property can be specified in the `scripts` section of the YAML file as a command argument to programs with dedicated flags. See the details [here](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/app-store-connect/fetch-signing-files.md#--issuer-idissuer_id). In that case, the environment variables will be fallbacks for missing values in scripts.
+{{</notebox>}}
+
+Once you have the App Store Connect API access set with mentioned above environment variables, you can get the build number using the tool and set your incremented project version
+
+```bash
+export APP_STORE_CONNECT_PRIVATE_KEY=$(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode) # if you encrypted the file itself, not its content
+LATEST_BUILD_NUMBER=app-store-connect get-latest-app-store-build-number '1234567890' # The argument is your application's Apple ID
+agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
+```
+
+To use the latest build number from Testflight use a similar command
+
+```bash
+export APP_STORE_CONNECT_PRIVATE_KEY=$(echo $APP_STORE_CONNECT_PRIVATE_KEY | base64 --decode) # if you encrypted the file itself, not its content
+LATEST_BUILD_NUMBER=app-store-connect get-latest-testflight-build-number '1234567890' # The argument is your application's Apple ID
+agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
+```
```

Comment:
Do these steps apply to flutter UI or yaml? As you mention yaml below, I assume this applies to Flutter UI. Can you mention before the code samples that you should add these to the `Pre-build script`(is it correct?) custom build step ( and link to the [custom build steps](https://docs.codemagic.io/flutter/custom-scripts/).

I would consider adding a subsection `Incrementing build number in Flutter UI` after the `Saving the API key to environment variables` subsection.